### PR TITLE
renames diffsync.DiffSync to diffsync.Adapter

### LINKED ISSUE #233: `diffsync.DiffSync` naming
Since we are calling subclasses of `diffsync.DiffSync`adapters in the examples and even in the docs

> When you initialize the `Diffsync` Adapter class

maybe we should just rename the class to `diffsync.Adapter`? To me this is clearer.

### PR DESCRIPTION
**Pull request by Kircheneer**
Closes #233

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -130,16 +130,18 @@ jobs:
       fail-fast: true
       matrix:
         python-version: ["3.7", "3.8", "3.9", "3.10"]
+        poetry-version: ["1.5.1"]
```
Comment (chadell) [inline .github/workflows/ci.yml:133]: do you expect to parametrize test by poetry? why not pin it?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -130,16 +130,18 @@ jobs:
...
     runs-on: "ubuntu-20.04"
     env:
       PYTHON_VER: "${{ matrix.python-version }}"
     steps:
       - name: "Check out repository code"
         uses: "actions/checkout@v2"
       - name: "Setup environment"
-        uses: "networktocode/gh-action-setup-poetry-environment@v5"
+        uses: "networktocode/gh-action-setup-poetry-environment@3ea5d3e"
```
Comment (chadell) [inline .github/workflows/ci.yml:141]: is this a commit? shouldn't we use tags?

**Review by chadell — APPROVED** [review]
LGTM, my only question is on the test

**Comment by Kircheneer** [comment]
Sorry @chadell, I was still experimenting with CI breakage, I will set to draft for now

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Kircheneer) [reply .github/workflows/ci.yml:133]: This is the act of pinning it, without this its `latest`

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Kircheneer) [reply .github/workflows/ci.yml]: The latest change that allows me to pin the poetry version is not released

**Pull request merged**